### PR TITLE
alias and slug alternates for widgets too

### PR DESCRIPTION
**Pull request by giannik**
allow alias and slug alternates for widgets along woth content shapes

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Alias/Services/AliasShapeTableProvider.cs
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Alias/Services/AliasShapeTableProvider.cs:50]: Apart from the alternate names, this is the same code as above for `Content`. DRY it instead to a method. Same for `AutorouteShapeTableProvider`.

**Review by Piedone — CHANGES_REQUESTED** [review]
Thank you for your contribution! When addressing review feedback, please adhere to the following:

- Please update your pull request according to feedback until it is approved by one of the core team members.
- Apply [suggested changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes) directly so the reviewer doesn't have to eyeball the changes. These resolve themselves after applying them, and that's fine.
- Don't resolve other conversations so it's easier to track for the reviewer. Then, the reviewer will resolve them.
- Feel free to mark conversations that you addressed to keep track of them with an emoji or otherwise, just don't resolve them.
- Please keep conversations happening in line comments in those convos, otherwise, communication will be a mess. If you have trouble finding them, see [this video](https://github.com/OrchardCMS/OrchardCore/pull/14749#issuecomment-1917976028).
- When you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

File: src/OrchardCore.Modules/OrchardCore.Alias/Services/AliasShapeTableProvider.cs
Comment (giannik) [reply src/OrchardCore.Modules/OrchardCore.Alias/Services/AliasShapeTableProvider.cs]: done

**Review by Piedone — CHANGES_REQUESTED** [review]
Looks good. Please also document it similar to https://docs.orchardcore.net/en/latest/reference/modules/Templates/#content__alias__alias and https://docs.orchardcore.net/en/latest/reference/modules/Templates/#content__slug__slug, see https://docs.orchardcore.net/en/latest/guides/contributing/contributing-documentation/. Also add a note about it to the release notes of the upcoming release (you can find this in the docs/OrchardCore.Docs/releases folder).

**Comment by giannik** [comment]
@Piedone can you check the conflict with src/docs/releases/2.1.0.md

**Comment by Piedone** [comment]
What do you mean by check the conflict? Please merge from `main` and resolve the conflict. Otherwise, the PR looks good.

**Comment by giannik** [comment]
replaced by #16860

**Comment by Piedone** [comment]
Please merge next time instead of opening a new PR. This adds needless friction, since I now need to review the whole PR again, not just the recent changes.

**Pull request closed without merging**
